client/my-sites/themes/index.web.js

```diff
@@ -24,26 +25,67 @@ export default function( router ) {
 				);
 			}
 			router(
-				`/themes/:vertical(${ verticals })?/:tier(free|premium)?`,
-				siteSelection, multiSite, makeNavigation, makeLayout
+				`/themes/:tier(free|premium)?/:site_id(${ siteId })?`,
+				siteSelection,
+				loggedIn,
+				makeNavigation,
+				makeLayout
 			);
 			router(
-				`/themes/:vertical(${ verticals })?/:tier(free|premium)?/:site_id`,
-				siteSelection, singleSite, makeNavigation, makeLayout
+				`/themes/:tier(free|premium)?/filter/:filter/:site_id(${ siteId })?`,
+				fetchThemeFilters,
+				validateFilters,
+				siteSelection,
+				loggedIn,
+				makeNavigation,
+				makeLayout
 			);
 			router(
-				`/themes/:vertical(${ verticals })?/:tier(free|premium)?/filter/:filter`,
-				validateFilters, siteSelection, multiSite, makeNavigation, makeLayout
+				`/themes/:vertical?/:tier(free|premium)?/:site_id(${ siteId })?`,
+				fetchThemeFilters,
+				validateVertical,
+				siteSelection,
+				loggedIn,
+				makeNavigation,
+				makeLayout
```

Comment:
Let's definitely collapse route defs with identical sets of middlewares into one.